mdx-web/src/main/java/com/mx/path/model/mdx/web/filter/RequireAuthenticationFilter.java

```diff
@@ -46,6 +47,12 @@ protected final void doFilterInternal(HttpServletRequest request, HttpServletRes
       throw new UnauthorizedException("Not Authenticated", "Not Authenticated");
     }
 
+    if (PathTools.isUserRequiredPath(path)) {
+      String userId = PathTools.extractUserId(path);
+      if (userId != null && Session.current().getUserId() != null && !userId.equals(Session.current().getUserId())) {
```

Comment:
I think this can be simplified a little:

```suggestion
      if (userId != null && !userId.equals(Session.current().getUserId())) {
```

Reply:
We should be throwing an exception if a userId is passed in on a user route and there isn't a userId in the session, right? Currently this doesn't throw an exception in that case

Reply:
Yes. That would be an unauthenticated session. No calls with the user id should be attempted until the session authentication is complete. This may be check somewhere else, but I"m not sure.

Reply:
Test has been added verifying we now throw a 401 when an authenticated session is missing a session userId on a user required route.

Also rebased onto master w/ latest changes